src/main/java/com/easypost/http/Requestor.java

```diff
@@ -419,11 +412,29 @@ private static EasyPostResponse makeURLConnectionRequest(final RequestMethod met
      * @param params The params of the API request.
      * @param clazz  The class of the object for deserialization
      * @param client The EasyPostClient object.
-     * @return A class object.
-     * @throws EasyPostException when the request fails.
+     * @return A clazz-type object.
+     * @throws HttpError               when the HTTP connection cannot be made.
+     * @throws EncodingError           when the request query cannot be encoded.
+     * @throws JsonError               when the request body cannot be encoded.
+     * @throws RedirectError           when the request is redirected.
+     * @throws UnauthorizedError       when the request is unauthorized.
+     * @throws ForbiddenError          when the request is forbidden.
+     * @throws PaymentError            when the request requires payment.
+     * @throws NotFoundError           when the request endpoint is not found.
+     * @throws MethodNotAllowedError   when the request method is not allowed.
+     * @throws TimeoutError            when the request times out.
+     * @throws InvalidRequestError     when the request is invalid.
+     * @throws RateLimitError          when the request exceeds the rate limit.
+     * @throws InternalServerError     when the request fails due to an internal server error.
+     * @throws ServiceUnavailableError when the request fails due to a service unavailability.
+     * @throws GatewayTimeoutError     when the request fails due to a gateway timeout.
+     * @throws UnknownApiError         when the request fails due to an unknown API error.
      */
     public static <T> T request(final RequestMethod method, String url, final Map<String, Object> params,
-                                final Class<T> clazz, final EasyPostClient client) throws EasyPostException {
+                                final Class<T> clazz, final EasyPostClient client)
+            throws EncodingError, JsonError, RedirectError, UnauthorizedError, ForbiddenError, PaymentError,
+            NotFoundError, MethodNotAllowedError, TimeoutError, InvalidRequestError, RateLimitError,
+            InternalServerError, ServiceUnavailableError, GatewayTimeoutError, UnknownApiError, HttpError {
```

Comment:
Why was this change? `EasyPostException` should be sufficed

Reply:
An attempt at being more explicit about what exceptions get thrown by this method, and explanations about why in the docstrings. These are all sub-types of EasyPostException, so we certainly could keep it to just EasyPostException. Like I said, just trying to be explicit so the end user knows what can be thrown, in case they want to handle different exceptions differently (e.g. simply try again on a RateLimitError, but know something is wrong for a InvalidRequestError).

Thoughts?

Reply:
If we already include all the possible exceptions in the Javadoc, then do we really need to include them again in the method? It just seems too much to me, won't block this.